src/controllers/engine/controllerengine.cpp

```diff
@@ -223,6 +229,51 @@ void ControllerEngine::uninitializeScriptEngine() {
     }
 }
 
+void ControllerEngine::loadModule(QFileInfo moduleFileInfo) {
+    m_moduleFileInfo = moduleFileInfo;
+
+    QJSValue mod = m_pScriptEngine->importModule(moduleFileInfo.absoluteFilePath());
+    if (mod.isError()) {
+        showScriptExceptionDialog(mod);
+        return;
+    }
+
+    connect(&m_scriptWatcher,
+            &QFileSystemWatcher::fileChanged,
+            this,
+            &ControllerEngine::scriptHasChanged);
+    m_scriptWatcher.addPath(moduleFileInfo.absoluteFilePath());
+
+    QJSValue initFunction = mod.property("init");
+    executeFunction(initFunction, QJSValueList{});
+
+    QJSValue handleInputFunction = mod.property("handleInput");
```

Comment:
Why not stick with the term "incomingData"? 
Else I would like to emphasis that it is a catch all callback. "allIncommingData" or similar.

Reply:
Because this is a completely new system.